--- orig/apache-cassandra-3.7/conf/cassandra-env.sh	2016-09-06 16:29:11.982958292 +0000
+++ hdeploy/apache-cassandra-3.7/conf/cassandra-env.sh	2016-09-06 16:33:37.437066342 +0000
@@ -56,13 +56,14 @@
     # pick the max
     half_system_memory_in_mb=`expr $system_memory_in_mb / 2`
     quarter_system_memory_in_mb=`expr $half_system_memory_in_mb / 2`
-    if [ "$half_system_memory_in_mb" -gt "1024" ]
+### HDEPLOY CHANGE: cap to 512MB
+    if [ "$half_system_memory_in_mb" -gt "512" ]
     then
-        half_system_memory_in_mb="1024"
+        half_system_memory_in_mb="512"
     fi
-    if [ "$quarter_system_memory_in_mb" -gt "8192" ]
+    if [ "$quarter_system_memory_in_mb" -gt "512" ]
     then
-        quarter_system_memory_in_mb="8192"
+        quarter_system_memory_in_mb="512"
     fi
     if [ "$half_system_memory_in_mb" -gt "$quarter_system_memory_in_mb" ]
     then
